CA1416 warning: Console.CursorVisible.get only supported on Windows

The following warning may appear when trying to build the project:

```
/home/christopher/Development/GitHub/Xyaneon/Weapons-of-Mass-Domination/src/Xyaneon.Console.Menus/Menu.cs(96,33): warning CA1416: This call site is reachable on all platforms. 'Console.CursorVisible.get' is only supported on: 'windows'. [/home/christopher/Development/GitHub/Xyaneon/Weapons-of-Mass-Domination/src/Xyaneon.Console.Menus/Xyaneon.Console.Menus.csproj]
```

Tring to launch the console app on Ubuntu also results in a crash on the title screen, right after the logo display:

```
Unhandled exception. System.PlatformNotSupportedException: Operation is not supported on this platform.
   at System.ConsolePal.get_CursorVisible()
   at System.Console.get_CursorVisible()
   at Xyaneon.Console.Menus.Menu.Run() in /home/christopher/Development/GitHub/Xyaneon/Weapons-of-Mass-Domination/src/Xyaneon.Console.Menus/Menu.cs:line 96
   at Program.<Main>$(String[] args) in /home/christopher/Development/GitHub/Xyaneon/Weapons-of-Mass-Domination/src/wmd-console/Program.cs:line 7
```

A more cross-platform compatible alternative should be explored.

--- a/src/Xyaneon.Console.Menus/Menu.cs
+++ b/src/Xyaneon.Console.Menus/Menu.cs
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.InteropServices;
 
 namespace Xyaneon.Console.Menus;
 
@@ -93,8 +94,12 @@ public class Menu
 
         NavigateTo(Pages.First());
 
-        bool cursorWasVisible = System.Console.CursorVisible;
-        System.Console.CursorVisible = false;
+        bool cursorWasVisible = true;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            cursorWasVisible = System.Console.CursorVisible;
+            System.Console.CursorVisible = false;
+        }
 
         while (!HasClosed)
         {
@@ -138,7 +143,11 @@ public class Menu
                 }
             }
         }
-        System.Console.CursorVisible = cursorWasVisible;
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            System.Console.CursorVisible = cursorWasVisible;
+        }
     }
 
     public void SetResultAndClose(object? result)

--- a/src/wmd-console/UI/PrintingUtility.cs
+++ b/src/wmd-console/UI/PrintingUtility.cs
@@ -1,4 +1,5 @@
-﻿using WMD.Game.State.Data;
+﻿using System.Runtime.InteropServices;
+using WMD.Game.State.Data;
 
 namespace WMD.Console.UI;
 
@@ -155,7 +156,9 @@ static class PrintingUtility
 
     public static void SetOutputEncoding()
     {
-        System.Console.OutputEncoding = System.Text.Encoding.Unicode;
+        System.Console.OutputEncoding = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+            ? System.Text.Encoding.Unicode
+            : System.Text.Encoding.UTF8; // System.Text.Encoding.GetEncoding(28591);
     }
 
     public static void PrintTitle()

--- a/src/wmd-console/UI/PrintingUtility.cs
+++ b/src/wmd-console/UI/PrintingUtility.cs
@@ -158,7 +158,7 @@ static class PrintingUtility
     {
         System.Console.OutputEncoding = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
             ? System.Text.Encoding.Unicode
-            : System.Text.Encoding.UTF8; // System.Text.Encoding.GetEncoding(28591);
+            : System.Text.Encoding.UTF8;
     }
 
     public static void PrintTitle()

